nanocloud/routes/users/users.go

```diff
@@ -128,18 +128,28 @@ func Disable(userId string) (int, error) {
 
 func Update(c *echo.Context) error {
 	u := users.User{}
-
+	uid := c.Param("id")
+	u.SetID(uid)
 	err := utils.ParseJSONBody(c, &u)
 	if err != nil {
-		return nil
+		return apiErrors.InvalidRequest
 	}
 
 	user, err := users.GetUser(u.GetID())
 	if err != nil {
 		return apiErrors.UserNotFound
 	}
 
-	if u.Password != "" {
+	if u.IsAdmin != user.IsAdmin {
+		if u.Id == user.Id {
+			return apiErrors.Unauthorized.Detail("You can't change your own rank")
```

Comment:
Missing dot at the end of the message.
While you are at it, please change can't -> cannot